### PR TITLE
DATA-24 - Adding additional fields to report.

### PR DESCRIPTION
**Pull request by Squirrel18**
## Description:

This PR adds additional fields to existing report and also adds a new type of report data to process.

## Reviewers:

- [ ] @andrey-canon

### CODE REVIEW

File: openedx_proversity_reports/tasks.py [hunk 1]
```diff
@@ -38,41 +43,21 @@ def generate_completion_report(courses, block_report_filter):
...
+    """
+    report_data = {
+        'last_page_data': {},
+        'exit_count_data': {},
+    }
+
+    last_page_data = get_last_page_accessed_data(courses)
+    if last_page_data:
+        report_data['last_page_data'] = last_page_data
+        exit_count_data = get_exit_count_data(last_page_data, courses)
```
Comment (andrey-canon) [inline openedx_proversity_reports/tasks.py]: you can merge line 60 and 61

File: openedx_proversity_reports/reports/last_page_accessed.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+            parent_tree_name = ''
+
+            if last_completed_child_position:
+                vertical_blocks = blocks.topological_traversal(
+                    filter_func=lambda block_key: block_key.block_type == 'vertical',
+                    yield_descendants_of_unyielded=True,
+                )
+
+                for vertical in vertical_blocks:
+                    for problem in blocks.get_children(vertical):
```
Comment (andrey-canon) [inline openedx_proversity_reports/reports/last_page_accessed.py]: could you rename to component

File: openedx_proversity_reports/reports/last_page_accessed.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+                        if problem.block_id == last_completed_child_position.block_key.block_id:
+                            parent_tree = get_parent_tree(blocks, problem)
```
Comment (andrey-canon) [inline openedx_proversity_reports/reports/last_page_accessed.py]: get_parent_tree can be called inside get_parent_display_names

File: openedx_proversity_reports/reports/last_page_accessed.py [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+            courseenrollment__course_id=course_key,
+            courseenrollment__is_active=1,
+            is_staff=0,
+        )
+
+        if not enrolled_students:
+            continue
+
+        usage_key = get_modulestore().make_course_usage_key(course_key)
+        blocks = get_course_blocks(enrolled_students[0], usage_key)
```
Comment (andrey-canon) [inline openedx_proversity_reports/reports/last_page_accessed.py]: what happen if enrolled_students is empty, does this need a try except block ?

File: openedx_proversity_reports/reports/last_page_accessed.py [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+            is_staff=0,
+        )
+
+        if not enrolled_students:
+            continue
+
+        user_data = []
+
+        usage_key = get_modulestore().make_course_usage_key(course_key)
+        blocks = get_course_blocks(enrolled_students[0], usage_key)
```
Comment (andrey-canon) [inline openedx_proversity_reports/reports/last_page_accessed.py]: what happen if enrolled_students is empty, does this need a try except block ?

Comment (Squirrel18) [reply openedx_proversity_reports/reports/last_page_accessed.py]: Line 44 will catch it. @andrey-canon

File: openedx_proversity_reports/reports/last_page_accessed.py (hunk 4 above)
Comment (Squirrel18) [reply openedx_proversity_reports/reports/last_page_accessed.py]: Line 139 will catch it. @andrey-canon

File: openedx_proversity_reports/reports/last_page_accessed.py (hunk 2 above)
Comment (Squirrel18) [reply openedx_proversity_reports/reports/last_page_accessed.py]: Addressed.

File: openedx_proversity_reports/reports/last_page_accessed.py (hunk 3 above)
Comment (Squirrel18) [reply openedx_proversity_reports/reports/last_page_accessed.py]: Addressed.

**Comment by Squirrel18** [comment]
Comments addressed. @andrey-canon

File: openedx_proversity_reports/tasks.py (hunk 1 above)
Comment (Squirrel18) [reply openedx_proversity_reports/tasks.py]: Addressed.

**Pull request merged**